### PR TITLE
Accent colors!

### LINKED ISSUE #2579: Calendar tab bar differs from the rest
Android only. Honestly I like it better at the moment

### LINKED ISSUE #2580: Android Recent Apps color value not set
It looks kinda weird.

Hawken's Screenshot

We can apparently fix this by adding to a new file `android/app/src/main/res/values/colors.xml` the following XML, 

```xml
<?xml version="1.0" encoding="utf-8"?>
<resources>
    <!-- Change the color value for your need -->
    <color name="colorPrimary">#3F51B5</color>
    <color name="colorPrimaryDark">#303F9F</color>
    <color name="colorAccent">#FF4081</color>
</resources>
```

then also adding the following XML to `android/app/src/main/res/values/styles.xml`

```xml
<resources>
    <!-- Base application theme. -->
    <style name="AppTheme" parent="Theme.AppCompat.Light.NoActionBar">
        <!-- Customize your theme here. -->
        <item name="colorPrimary">@color/colorPrimary</item>
        <item name="colorPrimaryDark">@color/colorPrimaryDark</item>
        <item name="colorAccent">@color/colorAccent</item>
    </style>
</resources>
```

### PR DESCRIPTION
**Pull request by hawkrives**
This adds a concept of an `accent color` to the app, and uses it.

It also centralizes most of the app colors, for easier playing in the future.

The accent color is currently St. Olaf's "purple" color, `#6E3A5D`, on a white background, `#fff#.

Pltfm | Home | Menus | SIS | Settings |
-- | -- | -- | -- | -- |
iOS | ios-home | ios-menus | ios-sis | ios-settings
Android | android-home | android-menus | android-sis | android-settings

~~I will now try to get my Android emulator running for screenshots there.~~

Closes #2579 – I think that's already fixed, technically, but I _know_ it's consistent as of this branch.
Closes #2580 – uses gold for the app switcher header
Closes #2581 – completely replaced by 3b12339

### CODE REVIEW

**Review by rye — DISMISSED** [review]

**Review by drewvolz — DISMISSED** [review]

**Review by hannesmcman — DISMISSED** [review]
Looks good!

File: android/build.gradle [hunk 1]
```diff
@@ -6,7 +6,7 @@ buildscript {
         google()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.1.0-alpha09'
+        classpath 'com.android.tools.build:gradle:3.1.3'
```
Comment (rye) [inline android/build.gradle]: This seems out of scope to me? Oh well.

**Review by rye — DISMISSED** [review]

**Comment by drewvolz** [comment]
As far as accents go, do we want to replace specific elements with the accent color? For example, I'm on the radio views and want to know if the buttons should inherit the accent color. Essentially, buttons and push button cells (their text colors), too.

* [x] Button colors
* [ ] Push-button cell colors
* [x] Enabled switch colors
* [ ] ~~Grouped table cell title colors?~~
* [ ] ~~Card title colors?~~

File: android/build.gradle (hunk 1 above)
Comment (drewvolz) [reply android/build.gradle]: I felt that way, too. I won't ask for it to be pulled out, but it would certainly be nice to have it as a separate PR.

Comment (hawkrives) [reply android/build.gradle]: Android Studio really likes to change it. I can pull it out.

**Comment by hawkrives** [comment]
@drewvolz Radio's buttons actually currently are supposed to inherit from the station's logo color; I'd be down for changing the rest of the denim-colored buttons, though (in Important Contacts and whatnot).

**Comment by drewvolz** [comment]
I've now added a list of four things I've been thinking about to my original message.

**Comment by hawkrives** [comment]
I don't think we should style ListSectionHeaders on iOS. Not even Calendar or Music do. 


I don't know what you mean by "button / push button".  

I don't really want to change iOS Switch colors.  They just don't feel right if they're anything but green to me. 

I feel that card titles are in the same boat as list section headers; they shouldn't be colored, at least in their current incarnation.

**Comment by drewvolz** [comment]
> I don't think we should style ListSectionHeaders on iOS. Not even Calendar or Music do.

Okay

> I don't know what you mean by "button / push button".

Detail views have `<Button />` components. They're styled to a blue at the moment. Things like the buttons in the Help view, or Contacts, etc.

We also have `<PushButtonCell />` components which are table cells that are buttons. Things like the submit button on building hours suggestions, etc.

> I don't really want to change iOS Switch colors. They just don't feel right if they're anything but green to me.

So EditHome is an exception to that? We are coloring the Android switches which is why I asked about iOS.

> I feel that card titles are in the same boat as list section headers; they shouldn't be colored, at least in their current incarnation.

So...

> in their current incarnation

Colored black?

**Comment by hawkrives** [comment]
I'll buy the PushButtonCell's getting the accent color. 

I can do Button's getting the accent color, although I will probably not change the radio ones. 

I'll do Switches, as well, at least to see how I feel about them. 

Black is not a color.

**Comment by drewvolz** [comment]
> Black is not a color.

Do you really want to pick on the word shaded vs colored? I feel like that's sidetracking.

**Comment by hawkrives** [comment]
I don't feel that they should acquire a hue. IMHO, they should remain black.

**Comment by drewvolz** [comment]
Would you like help changing any of those?

**Comment by hawkrives** [comment]
Current status:

* [x] Button colors
* [x] Push-button cell colors*
* [x] Enabled switch colors
* [ ] ~~Grouped table cell title colors?~~
* [ ] ~~Card title colors?~~

*: we don't have any "link-like" push-button cells, which are the ones Apple colorizes (like the footer items in Maps, or the Reset items in Settings). All of our push-button cells are currently view-pushing ones, which don't get colorized.

I've colorized the buttons, though:

ios | android
--- | ---
screen shot 2018-08-04 at 2 05 49 pm | 


I've also automated all of the "readable color" construction, so we'll be accessible according to WCAG AA-small guidelines whenever we adjust the themes.

**Review by drewvolz — DISMISSED** [review]
👍 :shipit:

**Review by drewvolz — APPROVED** [review]
I updated the tests and screenshots. 👍 :shipit:

**Review by rye — APPROVED** [review]
